src/integrations/admin/indexing-notification-integration.php

```diff
@@ -147,16 +152,15 @@ public function register_hooks() {
 			\add_action( 'admin_init', [ $this, 'cleanup_notification' ] );
 		}
 
-		if ( $this->options_helper->get( 'indexing_reason' ) || ! \wp_next_scheduled( self::NOTIFICATION_ID ) ) {
-			\add_action( 'admin_init', [ $this, 'create_notification' ] );
-		}
-
 		if ( ! \wp_next_scheduled( self::NOTIFICATION_ID ) ) {
 			\wp_schedule_event( $this->date_helper->current_time(), 'daily', self::NOTIFICATION_ID );
-			return;
+			\add_action( self::NOTIFICATION_ID, [ $this, 'create_notification' ] );
+			$this->options_helper->set( 'indexing_reason', self::REASON_FIRST_INSTALL );
```

Comment:
I think if you set this in `_wpseo_activate` ( right after tracking is set ) then you don't need any changes here except for the `REASON_FIRST_INSTALL` const being added here.

Reply:
that would add a dependency on indexing-notification-integration - we've hardcoded the reason for now